src/astro/sql/tables.py

```diff
@@ -0,0 +1,55 @@
+import random
+import string
+from dataclasses import dataclass
+from typing import List, Optional, Union
+
+from sqlalchemy import Column, MetaData
+
+MAX_TABLE_NAME_LENGTH = 63
+
+
+@dataclass
+class Metadata:
+    schema: Union[str, None] = None
+    database: Union[str, None] = None
+    warehouse: Union[str, None] = None
+    role: Union[str, None] = None
+
+
+@dataclass
+class Table:
+    # TODO: discuss alternative names to this class, since it contains metadata as opposed to be the
+    # SQL table itself
+    # Some ideas: TableRef, TableMetadata, TableData, TableDataset
```

Comment:
I think the hard part is that this class is front facing, so whatever we name it is gonna be something that users see. This might be something to discuss with Mike